nntrainer/layers/dropout.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file   dropout_layer.cpp
+ * @date   16 June 2020
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug	   No known bugs except for NYI items
+ * @brief  This is Dropout Layer Class for Neural Network
+ *
+ */
+
+#include <dropout.h>
+#include <layer_internal.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int DropOutLayer::initialize(Manager &manager) {
+  output_dim = input_dim;
+  // TODO : asking Tensor to manager
+  for (auto &t : input_dim) {
+    mask.push_back(std::make_shared<Tensor>(t, true));
+  }
+
+  return ML_ERROR_NONE;
+}
+
+void DropOutLayer::forwarding(bool training) {
+  auto &rate_ = std::get<props::DropOutSpec>(dropout_rate).get();
```

Comment:
@zhoonit  can this throw if rate_ is not set by the user?

Reply:
yes. if empty it throws

Reply:
Then perhaps this should be check at initialize phase for all the necessary properties like DropOutSpec here?

Reply:
it'd better be but not necessarily I guess.

Reply:
i guess this can be verified with layerSemantics tests later, where not setting mandatory properties must fail initialize/finalize.

Reply:
I guess this discussion went a little off-topic for this PR, I will resolve this conversation.